playpen/middleman/server.py

```diff
@@ -0,0 +1,368 @@
+        )
+
+    # spawning_result: Optional[SpawningResult] = None
+
+
+@app.add_request(method="getCodeplanAgentSolution")
+def get_codeplan_agent_solution(
+    app: KaiRpcApplication,
+    server: JsonRpcServer,
+    id: JsonRpcId,
+    params: GetCodeplanAgentSolutionParams,
+) -> None:
+    # create a set of AnalyzerRuleViolations
+    # seed the task manager with these violations
+    # get the task with priority 0 and do the whole thingy
+
+    app.log.info(f"get_codeplan_agent_solution: {params}")
+    if not app.initialized:
+        server.send_response(id=id, error=ERROR_NOT_INITIALIZED)
+        return
+
+    app.config = cast(KaiRpcApplicationConfig, app.config)
+
+    try:
+        model_provider = ModelProvider(app.config.model_provider)
+    except Exception as e:
+        server.send_response(
+            id=id,
+            error=JsonRpcError(
+                code=JsonRpcErrorCode.InternalError,
+                message=str(e),
+            ),
+        )
+        return
+
+    # Data for AnalyzerRuleViolation should probably take an ExtendedIncident
+    seed_tasks: list[Task] = []
+
+    for incident in params.incidents:
+        seed_tasks.append(
+            AnalyzerRuleViolation(
+                file=urlparse(incident.uri).path,
+                line=incident.line_number,
+                column=-1,  # Not contained within report?
+                message=incident.message,
+                priority=0,
+                incident=Incident(**incident.model_dump()),
+                violation=Violation(
+                    description=incident.violation_description or "",
+                ),
+                ruleset=RuleSet(
+                    name=incident.ruleset_name,
+                    description=incident.ruleset_description or "",
+                ),
+            )
+        )
+
+    rcm = RepoContextManager(
+        project_root=app.config.root_path,
+        reflection_agent=ReflectionAgent(
+            llm=model_provider.llm, iterations=1, retries=3
+        ),
+    )
+
+    server.send_notification(
+        "gitVFSUpdate",
+        GitVFSUpdateParams.from_snapshot(rcm.first_snapshot).model_dump(),
+    )
+
+    task_manager_config = RpcClientConfig(
+        repo_directory=app.config.root_path,
+        analyzer_lsp_server_binary=app.config.analyzer_lsp_rpc_path,
+        rules_directory=app.config.analyzer_lsp_rpc_path / "rules",
+        analyzer_lsp_path=app.config.analyzer_lsp_path,
+        analyzer_java_bundle_path=app.config.analyzer_lsp_path / "java_bundle",  # ?
+        label_selector=None,
+        incident_selector=None,
+        included_paths=None,
+    )
+
+    task_manager = TaskManager(
+        config=task_manager_config,
+        rcm=rcm,
+        seed_tasks=seed_tasks,
+        validators=[
+            MavenCompileStep(task_manager_config),
+            AnalyzerLSPStep(task_manager_config),
+        ],
+        agents=[
```

Comment:
We are going to want to add the dependency agents

*note to selves, rename to task runners*